src/components/interactive-map.js

```diff
@@ -151,20 +149,24 @@ export default class InteractiveMap extends PureComponent {
 
   componentDidMount() {
     const {eventCanvas} = this.refs;
-    const {mapControls} = this.props;
-
-    // Register event handlers defined by map controls
-    const events = {};
-    mapControls.events.forEach(eventName => {
-      events[eventName] = this._handleEvent;
-    });
+    const mapControls = this.props.mapControls || new MapControls();
```

Comment:
We used to set a default value for the prop `mapControls`. The change introduced by this PR makes the state of `mapControls` dependent on `eventManager` - when the component mounts, a new `eventManager` is created, and `mapControls` must be reset.

This new behavior is potentially introducing bugs if the user is not managing their custom `mapControls` instance properly. We used to call this prop `MapControls` which expects a constructor. Any suggestions?